### PR TITLE
Refactor : refactor the IT to support customized proxy test image.

### PR DESCRIPTION
**Pull request by taojintianxia**
---

Before committing this PR, I'm sure that I have checked the following options:
- [x] My code follows the [code of conduct](https://shardingsphere.apache.org/community/en/involved/conduct/code/) of this project.
- [x] I have self-reviewed the commit code.
- [x] I have triggered maven check: `mvn clean install -B -T2C -DskipTests -Dmaven.javadoc.skip=true -e`.
- [x] I have made corresponding changes to the documentation.
- [x] I have added corresponding unit tests for my changes.

### CODE REVIEW

File: shardingsphere-test/shardingsphere-integration-test/shardingsphere-integration-test-env/src/test/java/org/apache/shardingsphere/test/integration/env/container/atomic/util/AdapterContainerUtil.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+@NoArgsConstructor(access = AccessLevel.PRIVATE)
+public class AdapterContainerUtil {
+    
+    /**
+     * Get adapter container image.
+     * 
+     * @return adapter container image
+     */
+    public static String getAdapterContainerImage() {
+        String result = System.getProperty("it.docker.proxy.image");
```
Comment (TeslaCN) [inline shardingsphere-test/shardingsphere-integration-test/shardingsphere-integration-test-env/src/test/java/org/apache/shardingsphere/test/integration/env/container/atomic/util/AdapterContainerUtil.java:36]: Default value could be specified in `System.getProperty("prop", "defaultValue")`.

Comment (taojintianxia) [reply shardingsphere-test/shardingsphere-integration-test/shardingsphere-integration-test-env/src/test/java/org/apache/shardingsphere/test/integration/env/container/atomic/util/AdapterContainerUtil.java:36]: got it.

File: shardingsphere-test/shardingsphere-integration-test/shardingsphere-integration-test-scaling/src/test/java/org/apache/shardingsphere/integration/data/pipeline/framework/container/compose/DockerContainerComposer.java [hunk 2]
```diff
@@ -65,7 +65,7 @@ public DockerContainerComposer(final DatabaseType databaseType, final String sto
             storageContainerConfig = StorageContainerConfigurationFactory.newInstance(databaseType);
         }
         storageContainer = getContainers().registerContainer((DockerStorageContainer) StorageContainerFactory.newInstance(databaseType, storageContainerImage,
-                "", storageContainerConfig));
+                "", StorageContainerConfigurationFactory.newInstance(databaseType)));
```
Comment (sandynz) [inline shardingsphere-test/shardingsphere-integration-test/shardingsphere-integration-test-scaling/src/test/java/org/apache/shardingsphere/integration/data/pipeline/framework/container/compose/DockerContainerComposer.java:68]: Seems it should use `storageContainerConfig`, but not create another instance

**Review by sandynz — CHANGES_REQUESTED** [review]

File: shardingsphere-test/shardingsphere-integration-test/shardingsphere-integration-test-scaling/src/test/java/org/apache/shardingsphere/integration/data/pipeline/framework/container/compose/DockerContainerComposer.java (hunk 2 above)
Comment (azexcy) [inline shardingsphere-test/shardingsphere-integration-test/shardingsphere-integration-test-scaling/src/test/java/org/apache/shardingsphere/integration/data/pipeline/framework/container/compose/DockerContainerComposer.java:68]: create mysql 8 container will volumn another `my.cnf`, relation code is at 61~64 line at this file

Comment (taojintianxia) [reply shardingsphere-test/shardingsphere-integration-test/shardingsphere-integration-test-scaling/src/test/java/org/apache/shardingsphere/integration/data/pipeline/framework/container/compose/DockerContainerComposer.java]: fixed

Comment (taojintianxia) [reply shardingsphere-test/shardingsphere-integration-test/shardingsphere-integration-test-scaling/src/test/java/org/apache/shardingsphere/integration/data/pipeline/framework/container/compose/DockerContainerComposer.java]: fixed

**Review by sandynz — APPROVED** [review]

**Pull request merged**